tests/test_arrange.py

```diff
@@ -49,77 +51,85 @@ def test_CleanInfilepath_Underscores_Match():
 
-        shutil.rmtree(tmp_dir)
+    # define old dir
+    old_dir = f"{ASSETS_DIR}/valid_dataset/{version}" # corresponding unzipped dir
 
+    # define and populate new dir
+    if os.path.exists(f"{ASSETS_DIR}/tmp"):
+        shutil.rmtree(f"{ASSETS_DIR}/tmp")
+    os.mkdir(f'{ASSETS_DIR}/tmp')
+    new_dir = arrange.prepare_dcm2niix_input(infile, False, f"{ASSETS_DIR}/tmp")
 
-def test_PrepareDcm2niixInput_ZipArchive_CatchNestedInputError():
+    # get info on two directories: old and new
+    file_set_old, file_tree_old, file_name_path_dict_old = arrange.tally_files(old_dir)
+    file_set_new, file_tree_new, file_name_path_dict_new = arrange.tally_files(new_dir)
 
-    tmp_dir = f"{ASSETS_DIR}/tmp"
-    shutil.os.mkdir(tmp_dir)
-    infile = f"{ASSETS_DIR}/dicom_nested_error.zip"
+    # check that the two sets of filenames (leaves, not considering nesting level) match
+    assert len(file_set_new - file_set_old) == 0
+    assert len(file_set_old - file_set_new) == 0
 
-    with pytest.raises(SystemExit) as exception:
-        arrange.prepare_dcm2niix_input(infile, False, tmp_dir)
+    # file-level
+    for file in [file_ for file_ in file_name_path_dict_new.keys() if ((file_ != '.DS_Store') & (file_[0:2] != '._'))]:
 
-    shutil.rmtree(tmp_dir)
+        # compare old and new file paths, check new one has expected form
+        old_file_path = file_name_path_dict_old[file]
+        new_file_path_1 = os.path.join(new_dir, file)
+        new_file_path_2 = file_name_path_dict_new[file]
+        print(f"old_file_path: {old_file_path}")
+        print(f"new_file_path_1: {new_file_path_1}")
+        print(f"new_file_path_2: {new_file_path_2}")
+        assert new_file_path_2 == new_file_path_1
 
-    assert exception.type == SystemExit
+        # check analogous files match
+        out = filecmp.cmp(old_file_path, new_file_path_2, shallow=True)
+        assert out
 
 
-def test_PrepareDcm2niixInput_TarArchive_MatchValidDataset():
+@pytest.mark.parametrize("version, ext", [
+    ("dicom_nested_one_level_collision", "zip"),
+    ("dicom_nested_two_levels_collision", "zip"),
+    ("dicom_nested_uneven_collision", "zip")
+])
+def test_PrepareDcm2niixInput_Archive_CatchCollisionError(version, ext):
 
-    versions = ["dicom_nested", "dicom_single", "parrec_single"]
-
-    for version in versions:
-
-        tmp_dir = f"{ASSETS_DIR}/tmp"
-        shutil.os.mkdir(tmp_dir)
-        infile = f"{ASSETS_DIR}/{version}.tgz"
-
-        arrange.prepare_dcm2niix_input(infile, False, tmp_dir)
-        valid_dir = f"{ASSETS_DIR}/valid_dataset/{version}"
-        out = filecmp.dircmp(tmp_dir, valid_dir)
-
-        assert out.left_only == []
-        assert out.right_only == []
-
-        shutil.rmtree(tmp_dir)
-
-
-def test_PrepareDcm2niixInput_TarArchive_CatchNestedInputError():
+    # source
+    infile = f"{ASSETS_DIR}/{version}.{ext}"
 
+    # target
     tmp_dir = f"{ASSETS_DIR}/tmp"
+    if os.path.exists(tmp_dir):
+        shutil.rmtree(tmp_dir)
```

Comment:
Let's use `tmpdir` fixture for this as well.

Reply:
👍